src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -112,7 +112,11 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
     }
 
     [UnconditionalSuppressMessage("Trimmer", "IL2026",
-        Justification = "We surface a RequireUnreferencedCode in the call to the Map method adding this EndpointDataSource. The trimmer is unable to infer this.")]
+        Justification = "We surface a RequireUnreferencedCode in the call to the Map methods adding route handlers to this EndpointDataSource. Analysis is unable to infer this. " +
+        "Map methods that configure a RequestDelegate don't use trimmer unsafe features.")]
+    [UnconditionalSuppressMessage("AOT", "IL3050",
+        Justification = "We surface a RequiresDynamicCode in the call to the Map methods adding route handlers this EndpointDataSource. Analysis is unable to infer this. " +
+        "Map methods that configure a RequestDelegate don't use AOT unsafe features.")]
```

Comment:
This method is pretty big, and I can imagine people adding to it over time.

Instead of suppressing these warnings for the whole method, can we extract just the 2 calls to `RequestDelegateFactory` into local methods, and then just suppress there? That way if someone adds new code that is AOT/trimming unsafe, they will get warnings.

Reply:
I tried that, and the logic is too intermingled. I didn't feel comfortable refactoring something I'm not that familiar. Can log improving this as a follow up task.